cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py

```diff
@@ -0,0 +1,475 @@
+    return False
+
+
+def verify_records_metrics_table(appliance, provider):
+    """Verify that rollups are present in the metric_rollups table."""
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    ems = appliance.db.client['ext_management_systems']
+    metrics = appliance.db.client['metrics']
+
+    # Capture real-time C&U data
+    ret = appliance.ssh_client.run_rails_command(
+        "\"vm = Vm.where(:ems_id => {}).where(:name => {})[0];\
+        vm.perf_capture('realtime', 1.hour.ago.utc, Time.now.utc)\""
+        .format(provider.id, repr(vm_name)))
+    assert ret.success, "Failed to capture VM C&U data:".format(ret.output)
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(metrics.id)
+            .join(ems, metrics.parent_ems_id == ems.id)
+            .filter(metrics.capture_interval_name == 'realtime',
+            metrics.resource_name == vm_name,
+            ems.name == provider.name, metrics.timestamp >= date.today())
+        )
+
+    for record in appliance.db.client.session.query(metrics).filter(
+            metrics.id.in_(result.subquery())):
+        return any([record.cpu_usagemhz_rate_average,
+            record.cpu_usage_rate_average,
+            record.derived_memory_used,
+            record.net_usage_rate_average,
+            record.disk_usage_rate_average])
+    return False
+
+
+@pytest.fixture(scope="module")
+def resource_usage(vm_ownership, appliance, provider):
+    """Retrieve resource usage values from metric_rollups table.
+
+    Chargeback reporting is done on hourly and daily rollup values and not real-time values.So, we
+    are capturing C&U data and forcing hourly rollups by running commands through
+    the Rails console.
+    """
+    average_cpu_used_in_mhz = 0
+    average_memory_used_in_mb = 0
+    average_network_io = 0
+    average_disk_io = 0
+    average_storage_used = 0
+    consumed_hours = 0
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    metrics = appliance.db.client['metrics']
+    rollups = appliance.db.client['metric_rollups']
+    ems = appliance.db.client['ext_management_systems']
+    logger.info('Deleting METRICS DATA from metrics and metric_rollups tables')
+
+    appliance.db.client.session.query(metrics).delete()
+    appliance.db.client.session.query(rollups).delete()
+
+    wait_for(verify_records_metrics_table, [appliance, provider], timeout=600,
+        message='Waiting for VM real-time data')
+
+    # New C&U data may sneak in since 1)C&U server roles are running and 2)collection for clusters
+    # and hosts is on.This would mess up our Chargeback calculations, so we are disabling C&U
+    # collection after data has been fetched for the last hour.
+
+    appliance.server.settings.disable_server_roles(
+        'ems_metrics_coordinator', 'ems_metrics_collector')
+
+    # Perfrom rollup of C&U data.
+    ret = appliance.ssh_client.run_rails_command(
```

Comment:
OPTIONAL: this call looks very similar to previous one. Can you combine them somehow ?
btw, if you could get rid of direct rails call, it would be even better.

Reply:
It cannot be combined.

I'll reach out to Dev asking them to provide this functionality through REST API.